spring-cloud-commons/src/main/java/org/springframework/cloud/commons/security/ResourceServerTokenRelayAutoConfiguration.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.commons.security;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.AllNestedConditions;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.security.oauth2.OAuth2AutoConfiguration;
+import org.springframework.boot.autoconfigure.security.oauth2.resource.UserInfoTokenServices;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+import org.springframework.security.oauth2.config.annotation.web.configuration.OAuth2ClientConfiguration;
+import org.springframework.security.oauth2.config.annotation.web.configuration.ResourceServerConfiguration;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+/**
+ * Adds an MVC interceptor for relaying OAuth2 access tokens into the client context (if
+ * there is one). In this way an incoming request to a resource server can be relayed
+ * downstream just be using <code>@EnableOAuth2Client</code> and an
+ * <code>OAuth2RestTemplate</code>. An MVC interceptor is used so as to have a minimal
+ * impact on the call stack. If you are not using MVC you could use a custom filter or AOP
+ * interceptor wrapping the same call to an {@link AccessTokenContextRelay}.
+ *
+ * <br/>
+ *
+ * N.B. an app that is using {@link UserInfoTokenServices} generally doesn't need this
+ * interceptor, but it doesn't hurt to include it.
+ *
+ * @author Dave Syer
+ *
+ */
+@Configuration(proxyBeanMethods = false)
+@AutoConfigureAfter(OAuth2AutoConfiguration.class)
+@ResourceServerTokenRelayAutoConfiguration.ConditionalOnOAuth2ClientInResourceServer
+@ConditionalOnClass(ResourceServerConfiguration.class)
+@ConditionalOnWebApplication
+@ConditionalOnProperty(value = "security.oauth2.resource.tokenRelay", matchIfMissing = true)
```

Comment:
we should reference properties via canonical form, so here `tokey-relay` (I know this was probably how it was).

I wonder if this property should be mvc specific. `spring.cloud.mvc.token-relay.enabled=true`? or something

Reply:
`spring.cloud.mvc.token-relay.enabled=true` sounds like a good idea

Reply:
that's even better cause those props will end up in the configprops.adoc since they start with spring.cloud

Reply:
In https://github.com/spring-cloud/spring-cloud-commons/blob/main/spring-cloud-commons/src/main/java/org/springframework/cloud/commons/security/ResourceServerTokenRelayAutoConfiguration.java

```
@ConditionalOnProperty(value = "spring.cloud.mvc.token-relay.enabled=true", matchIfMissing = true)
```

Is `=true` a mistake? the key includes `=true`?

Reply:
yes, fixed in f1b3956